Modules/Loadable/Sequences/Logic/vtkSlicerSequencesLogic.cxx

```diff
@@ -546,14 +598,28 @@ void vtkSlicerSequencesLogic::UpdateSequencesFromProxyNodes(vtkMRMLSequenceBrows
     {
     // Update sequence item from proxy node
     vtkMRMLSequenceNode* sequenceNode = browserNode->GetSequenceNode(proxyNode);
-    if (sequenceNode && browserNode->GetSaveChanges(sequenceNode) && browserNode->GetSelectedItemNumber()>=0)
+    if (sequenceNode && browserNode->GetSelectedItemNumber()>=0)
       {
-      std::string indexValue = masterNode->GetNthIndexValue(browserNode->GetSelectedItemNumber());
-      int closestItemNumber = sequenceNode->GetItemNumberFromIndexValue(indexValue, false);
-      if (closestItemNumber >= 0)
+      if (browserNode->GetSaveChanges(sequenceNode))
         {
-        std::string closestIndexValue = sequenceNode->GetNthIndexValue(closestItemNumber);
-        sequenceNode->UpdateDataNodeAtValue(proxyNode, indexValue, true /* shallow copy*/);
+        std::string indexValue = masterNode->GetNthIndexValue(browserNode->GetSelectedItemNumber());
+        // If missing item mode is MissingItemSetToDefault then it means that the current time point
+        // may not have a corresponding item in the sequence.
+        // In this case we should not update the closest item but only if there is an exact match.
```

Comment:
```suggestion
        // In this case, updating the closest item is only necessary when there is an
        // exact match; otherwise, it should be skipped with the proxy node 
        // modifications reverted.
```